### ISSUE TITLE
🐛 `files.include` not working (at all)

### ISSUE THREAD

**Issue by fubhy** [biomejs/biome#1117]
### Environment information

```block
pnpm biome rage
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.10.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

When specifying the `files.include` option in `biome.json`, it doesn't actually do anything.

The only way I managed to get `biome format` and `biome check` to work is by using `files.ignore` instead.

When using `files.ignore`, it doesn't matter if `files.include` is configured or not. It just doesn't appear to do anything.

However, if I don't specify `files.ignore`, then no files are checked (which is also weird). Imho, the natural expected behavior would be for `files.include` to widen the list and `files.ignore` to narrow it. And maybe, if no `files.include` is specified, include everything (or nothing, that's a matter of taste imho).

### Expected result

`files.include` should work I guess ;-)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by fubhy** [comment]
> Thank you @fubhy for filing the issue , could you please share a **minimal** reproduction, so we can investigate and help you? I'm aware that there are some issues, but we need a way to reproduce them.
> 
> As the issue stands , our hands are tight and we can't do anything because we don't have any concrete information to triage it.

I mean, as I said there's not much to set up in terms of a reproduction repo. Both `include` and `ignore` just don't work as documented as far as I can tell in the most simple combination even.

But here you go: https://github.com/fubhy/biome-broken-include-and-ignore

**Comment by fubhy** [comment]
Also please let me know if I'm doing sth. wrong here. If this is actually broken in the way that it appears to me it would actually be kinda wild. I mean considering there are no other issues open about it (as far as I can tell) for what seems to be should be a widely used configuration option.

**Comment by ematipico** [comment]
> Also please let me know if I'm doing sth. wrong here. If this is actually broken in the way that it appears to me it would actually be kinda wild considering that there are no other issues open about it (as far as I can tell) for what should(?) be a widely used configuration option.

For example, one thing that you didn't mention is **how** you're running Biome. Biome is CLI and LSP, and they work in different environments. The reproduction doesn't have any scripts, so I guess you're using Biome via LSP?

**Comment by fubhy** [comment]
Gotcha. I am seeing the issue when using the biome cli via e.g. `biome check .`

**Comment by ematipico** [comment]
> Gotcha. I am seeing the issue when using the biome cli via e.g. `biome check .`

Could you update the repository, please? Sorry for being peaky, doing `biome check .` could trigger a possible global installation and make things different

**Comment by fubhy** [comment]
> > Gotcha. I am seeing the issue when using the biome cli via e.g. `biome check .`
> 
> Could you update the repository, please? Sorry for being peaky, doing `biome check .` could trigger a possible global installation and make things different

No it doesn't in my case. I'm using a Nix based setup throughout my entire system. I'm directly calling `pnpm biome check .` in this case.

**Comment by anonrig** [comment]
I'm seeing a similar setup as well

**Comment by Conaclos** [comment]
I can confirm the bug with your reproduction. I tested with pnpm and npx:

```shell
$ pnpm biome check .
Checked 0 file(s) in 446µs
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.

$ npx biome check .
Checked 0 file(s) in 446µs
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
```

**Comment by jschuur** [comment]
I ran into a similar issue. I have this in `biome.json`:

```json
  "files": {
    "include": ["**/*.mjs", "**/*.js", "**/*.css", "**/*.json"]
  },
```

If I run `npx biome lint` I get `Missing argument <INPUT>`. I was expecting it to default to the include and ignore globs set up in the config file.

Curiously, if I run `npx biome lint .`, it checks 4 files, which happens to be the number of files matching the file extensions in the list above. It's as if it didn't go recursively into subdirectories, based on the `**` part of the path globs.
